--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   dones.png
	modified:   dones_final.png
	modified:   dones_time_final.png
	modified:   dones_x_time.png
	modified:   falling_penalty.png
	modified:   left_distance_reward.png
	modified:   obj_ori.png
	modified:   palm_facing.png
	modified:   pinky.png
	modified:   rewards.png
	modified:   rewards_mean.png
	modified:   setup_pip.sh
	modified:   source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env.py
	modified:   source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env_cfg.py
	modified:   success_reward.png
	modified:   time_reward.png
	modified:   total_reward.png

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/dones.png b/dones.png
index 9031c53..7b20480 100644
Binary files a/dones.png and b/dones.png differ
diff --git a/dones_final.png b/dones_final.png
index a79ad34..00c02e9 100644
Binary files a/dones_final.png and b/dones_final.png differ
diff --git a/dones_time_final.png b/dones_time_final.png
index cca7ce5..99257fc 100644
Binary files a/dones_time_final.png and b/dones_time_final.png differ
diff --git a/dones_x_time.png b/dones_x_time.png
index 28e0332..49d7e55 100644
Binary files a/dones_x_time.png and b/dones_x_time.png differ
diff --git a/falling_penalty.png b/falling_penalty.png
index e53d41a..ad51b45 100644
Binary files a/falling_penalty.png and b/falling_penalty.png differ
diff --git a/left_distance_reward.png b/left_distance_reward.png
index c0767d8..7c73a50 100644
Binary files a/left_distance_reward.png and b/left_distance_reward.png differ
diff --git a/obj_ori.png b/obj_ori.png
index a44007c..1f833e9 100644
Binary files a/obj_ori.png and b/obj_ori.png differ
diff --git a/palm_facing.png b/palm_facing.png
index f5e7424..4d29c08 100644
Binary files a/palm_facing.png and b/palm_facing.png differ
diff --git a/pinky.png b/pinky.png
index 2de1034..83d52ce 100644
Binary files a/pinky.png and b/pinky.png differ
diff --git a/rewards.png b/rewards.png
index 41c5871..52ae57a 100644
Binary files a/rewards.png and b/rewards.png differ
diff --git a/rewards_mean.png b/rewards_mean.png
index 2649e17..6e944d2 100644
Binary files a/rewards_mean.png and b/rewards_mean.png differ
diff --git a/setup_pip.sh b/setup_pip.sh
index e337f02..fccc675 100644
--- a/setup_pip.sh
+++ b/setup_pip.sh
@@ -1,4 +1,4 @@
-python3 -m pip install -e source/gr1_train
+python3 -m pip install -e /workspace/fourier-sim/gr1_train/source/gr1_train
 pip3 install matplotlib
 mv -f /workspace/isaaclab/_isaac_sim/kit/python/lib/python3.11/site-packages/rsl_rl/runners/on_policy_runner.py \
     /workspace/isaaclab/_isaac_sim/kit/python/lib/python3.11/site-packages/rsl_rl/runners/on_policy_runner_cpy.py
diff --git a/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env.py b/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env.py
index 1fddfeb..7927201 100644
--- a/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env.py
+++ b/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env.py
@@ -16,6 +16,8 @@ from isaaclab_assets.robots.fourier import GR1T2_HIGH_PD_CFG  # isort: skip
 
 from .gr1_train_env_cfg import Gr1TrainEnvCfg
 from isaaclab.sensors import ContactSensor
+from omni.physx import get_physx_scene_query_interface
+from pxr import Sdf, PhysicsSchemaTools
 
 import numpy as np
 import matplotlib.pyplot as plt
@@ -87,7 +89,7 @@ class Gr1TrainEnv(DirectRLEnv):
         # cache convenient handles
         self.joint_pos = self.robot.data.joint_pos
         self.joint_vel = self.robot.data.joint_vel
-    
+
     def close(self):
         total_reward = np.array(self.rewards["total_reward"])
         # rew_lift = np.array(self.rewards["rew_lift"])
diff --git a/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env_cfg.py b/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env_cfg.py
index 0b2a889..b367ca1 100644
--- a/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env_cfg.py
+++ b/source/gr1_train/gr1_train/tasks/direct/gr1_train/gr1_train_env_cfg.py
@@ -143,7 +143,7 @@ class Gr1TrainEnvCfg(DirectRLEnvCfg):
     # reward_scale_left_vel: float = -0.01
     # reward_scale_obj_vel: float = -0.7
     reward_scale_falling_penalty: float = -1000.0
-    reward_scale_contact_left_pinky: float = -2.0
+    reward_scale_contact_left_pinky: float = -20.0
     reward_object_orientation: float = -10.0
 
     reward_scale_time: float = -0.3
diff --git a/success_reward.png b/success_reward.png
index 540983b..445b9ab 100644
Binary files a/success_reward.png and b/success_reward.png differ
diff --git a/time_reward.png b/time_reward.png
index 69ee539..c1440ac 100644
Binary files a/time_reward.png and b/time_reward.png differ
diff --git a/total_reward.png b/total_reward.png
index 315e096..7dd0b51 100644
Binary files a/total_reward.png and b/total_reward.png differ